scriptlets/install_checkbox_agent_source

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+# Install Checkbox from repo, matching a specific version.
+#
+# Description:
+#
+# The script clones the Checkbox repo and then uses the `checkout_to_version`
+# tool in order to checkout and install a version of Checkbox that matches
+# a specific version provided as an argument.
+#
+# The second argument, the `tools_path` is necessary in order to locate the
+# `checkout_to_version` tool. It can be removed if that tool is installed (it
+# is currently not). The argument is also not required if TOOLS_PATH is set.
+
+usage() {
+    echo "Usage: $(basename $0) <version> [<tools_path>]"
+}
+
+if [ $# -gt 2 ]; then
+    usage
+    echo "Error: Too many arguments"
+    exit 1
+fi
+
+if [ $# -lt 1 ]; then
+    usage
+    echo "Error: <version> not specified"
+    exit 1
+fi
+
+VERSION=$1
+
+if [ $# -eq 2 ]; then
+    TOOLS_PATH=$2
+elif [ -z "$TOOLS_PATH" ]; then
+    usage
+    echo "Error: <tools_path> not specified and TOOLS_PATH not set"
+    exit 1
+fi
+
+# clone Checkbox repo
+git clone --filter=tree:0 https://github.com/canonical/checkbox.git > /dev/null
+
+# checkout commit to match the required version
+$TOOLS_PATH/version-published/checkout_to_version.py ~/checkbox "$VERSION"
+
+# install checkbox from source
+(cd checkbox/checkbox-ng; sudo python3 setup.py install > /dev/null)
```

Comment:
I think the original version of this ensured that checkbox gets installed in a virtualenv. Do we ensure that somehow?

Reply:
This is [corresponding code](https://github.com/canonical/hwcert-jenkins-jobs/blob/436cf72337a4bcfec2b42215361769d00477118f/jobs/sru/run-sru.sh#L325) in `run-sru.sh`:
```
hwcert-jenkins-tools/version-published/checkout_to_version.py ~/checkbox "\$CHECKBOX_VERSION"
(cd checkbox/checkbox-ng; sudo python3 setup.py install > /dev/null)
```
It's practically identical and I don't see a virtualenv being used. We can, of course, and that's why I wanted to offer an abstraction over the original code.

Reply:
that's true, I don't think we actually handle this yet. It does work ok for now, so we could defer it until it's an immediate problem. But we'll almost certainly have to resolve it at some point in the future.

Reply:
Using `pipx` (see comment below), this would be patched as below:
```
- (cd checkbox/checkbox-ng; sudo python3 setup.py install > /dev/null)
+ pipx install checkbox/checkbox-ng > /dev/null
```
I have tested this successfully in a multipass container.

Reply:
@plars I will install `pipx` explicitly and use it instead of `pip` here but do let me know if you think it might make sense to actually pre-install it when creating the container.

Reply:
It's even fine if we do both. Having it here is nice because we don't assume it's there and we ensure the requirements are installed. Having it in the pre-install means that installing it here will usually take no extra time.

Reply:
@plars `pipx` us not installed on the agent and used for installing `checkout_to_version`.